core/src/main/java/io/fabric8/maven/core/service/kubernetes/BuildahIntegration/BuildahBuildServiceUtil.java

```diff
@@ -0,0 +1,164 @@
+
+        log.info("Container %s successfully built and pushed.", con);
+
+        if(envList != null) {
+            buildah.config(con).env(envList).build().execute();
+        }
+
+        if(labelList != null) {
+            buildah.config(con).label(labelList).build().execute();
+        }
+
+        if(portSet != null) {
+            buildah.config(con).port(portSet).build().execute();
+        }
+
+        if(!entrypointList.isEmpty()) {
+            buildah.config(con).entrypoint(entrypointList).build().execute();
+        }
+
+        if (fatJar != null) {
+            String jarPath = targetDir + "/" ;
+            buildah.config(con).workingDir(jarPath).build().execute();
+            buildah.copy(con, String.valueOf(fatJar)).destination(jarPath).build().execute();
+        }
+
+        buildah.commit(con).withImageName(targetImage).build().execute();
+        buildah.rm().containerId(con).build().execute();
+        log.info("Image successfullly build with %s as target image.", targetImage);
+    }
+
+    public static BuildahBuildConfiguration getBuildahBuildConfiguration(BuildService.BuildServiceConfig config, BuildImageConfiguration buildImageConfiguration, String fullImageName, Logger log) throws MojoExecutionException {
+
+        io.fabric8.maven.docker.service.BuildService.BuildContext dockerBuildContext = config.getDockerBuildContext();
+        RegistryService.RegistryConfig registryConfig = dockerBuildContext.getRegistryConfig();
+
+        String targetDir = buildImageConfiguration.getAssemblyConfiguration().getTargetDir();
+
+        AuthConfig authConfig = registryConfig.getAuthConfigFactory()
+                .createAuthConfig(true, true, registryConfig.getAuthConfig(),
+                        registryConfig.getSettings(), null, registryConfig.getRegistry());
+
+        BuildahBuildConfiguration.Builder buildahBuildConfiguration = new BuildahBuildConfiguration.Builder(log).from(buildImageConfiguration.getFrom())
+                .ports(buildImageConfiguration.getPorts())
+                .targetImage(fullImageName)
+                .envMap(buildImageConfiguration.getEnv())
+                .labelMap(buildImageConfiguration.getLabels())
+                .buildDirectory(config.getBuildDirectory())
+                .targetDir(targetDir)
+                .entrypoint(buildImageConfiguration.getEntryPoint());
+
+        if (authConfig != null) {
+            buildahBuildConfiguration.authConfig(authConfig);
+        }
+
+        return buildahBuildConfiguration.build();
+    }
+
+    public static Path getFatJar(String buildDir, Logger log) {
+
+        FatJarDetector fatJarDetector = new FatJarDetector(buildDir);
+        try {
+            FatJarDetector.Result result = fatJarDetector.scan();
+            if (result != null) {
+                return result.getArchiveFile().toPath();
+            }
+
+        } catch (MojoExecutionException e) {
+            log.error("MOJO Execution exception occured: %s", e);
```

Comment:
typo, it might be from copy-paste; But Can we please change exception message to something else?